coalib/bearlib/languages/Language.py

```diff
@@ -410,40 +442,172 @@ def get_default_version(self):
         return type(self)(self.versions[-1]) if self.versions else type(self)()
 
 
-def limit_versions(language, limit, operator):
+def prepare_instance(language_, item):
+    """
+    It creates a new `Language` instance depending on the type of item. It is
+    used in __and__ , __or__ methods to avoid duplicate code.
+
+    :param item:
+        Item to use to create the new instance.
+    :return:
+        A `Language` instance.
+    """
+    if isinstance(item, float):
+        language = type(language_)((item))
+    elif isinstance(item, int):
+        language = Language[type(language_).__name__ + ' ' + str(item)]
+    elif isinstance(item, str):
+        language = Language[item]
+    else:
+        language = item
+    return language
+
+
+def limit_versions(language, specifier):
     """
-    Limits given languages with the given operator:
+    Limits the versions of a `Language` instance. It supports limiting by using
+    operators or specifying versions:
+
+    >>> Language['Python = 3']
```

Comment:
Better don't allow single `=`.  Either `==` or no op at all, using `~=` and adding `.0` in latter case